src/main/java/com/atlassian/db/replica/internal/state/ConnectionState.java

```diff
@@ -34,6 +34,27 @@ public final class ConnectionState {
     private final Warnings warnings;
     private final StateListener stateListener;
     private volatile boolean replicaConsistent = true;
+    /**
+     * When we use a connection to write to the database, it becomes a 'dirty' connection.
+     * The state is cleared when we commit the transaction. Currently, we use it only
+     * to detect dirty connection close() and commit the transaction before closing
+     * the connection.
+     * In the future we can use this state to exit `MainConnection` state and move more traffic to replicas.
```

Comment:
This line is interesting, I wonder whether an issue should not be created to don't miss it.
I can't imagine the moment when RR effectiveness it too slow and this would become a priority.

Reply:
At some point, we stopped to optimise the split. This way, the db-connection is less complex.